x-pack/plugins/global_search/server/types.ts

```diff
@@ -92,6 +93,11 @@ export interface GlobalSearchFindOptions {
    * If/when provided and emitting, no further result emission will be performed and the result observable will be completed.
    */
   aborted$?: Observable<void>;
+  /**
+   * A ES client of type IScopedClusterClient is passed to the `find` call.
+   * When performing calls to ES, the interested provider can utilize this parameter to identify the specific cluster.
+   */
+  client?: IScopedClusterClient;
```

Comment:
nitpicking here, but I think it's neater to pass the concrete Elasticsearch Client with permissions (so the result of `client.asCurrentUser`) instead of the cluster client which still needs to choose between `asCurrentUser` and `asInternalUser`, if possible?

Reply:
`get_index_data` expects a `ClusterClient` in [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/lib/indices/utils/get_index_data.ts#L103) and that is why I kept it as `clusterClient`. Should I write a different method which expects only `currentUser`?

Reply:
We should probably pass that function an ElasticsearchClient in all of its usages, but maybe that's too much effort for this PR.

Reply:
`fetch_indices` and `indices_search_results_providers*` both is using this function and probably their related test files. So not much, however, then we should update `getSearchIndexData` function and its usage too.

Is it okay if I create a separate small PR to handle this update for both functions?

Reply:
@sphilipse : Can I get an approval if everything looks good?